components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -278,7 +281,7 @@ private fun prepare(context: Context, preparers: List<IconPreprarer>, request: I
         preparer.prepare(context, preparedRequest)
     }
 
-private fun load(
+private suspend fun load(
```

Comment:
Tested more today and saw this method quote problematic because it's static and used by every call to load icons.
While the memory/disk loaders should be fast the http loader can be slow and I don't think it supports concurrent requests.
So one request that hangs might stall all future ones on this provider (until it timeouts).
@pocmo Just want to check if I'm following this ^^ right. If so this flow might need to be reconsidered.